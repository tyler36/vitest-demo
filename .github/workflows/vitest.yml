name: testing

on:
  workflow_dispatch:
  push:
    branches: [main]
  pull_request:
    branches: [main]

concurrency:
  group: ${{ github.head_ref || github.run_id }}
  cancel-in-progress: true

jobs:
  test:
    runs-on: ubuntu-latest

    permissions:
      # Required to checkout the code
      contents: read
      # Required to put a comment into the pull-request
      pull-requests: write

    steps:
    - name: '☁️ checkout repository'
      uses: actions/checkout@v4
      with:
        persist-credentials: false

    - name: '📦 Install Node'
      uses: actions/setup-node@v4
      with:
        node-version: '20.x'
        cache: 'npm'

    - name: '📦 Install Dependencies'
      run: npm ci

    - name: '✅ Run Vitest'
      run: npx vitest --coverage.enabled true

    - name: '📓 Report Coverage'
      # Set if: always() to also generate the report if tests are failing
      # Only works if you set `reportOnFailure: true` in your vite config as specified above
      if: always()
      uses:  davelosert/vitest-coverage-report-action@v2
      with:
        working-directory: './logs/'
