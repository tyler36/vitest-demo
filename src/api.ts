export async function fetchTodoList() {
  return (
    await fetch('https://jsonplaceholder.typicode.com/todos/1')
  ).json()
}
