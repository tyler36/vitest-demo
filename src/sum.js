export function sum(a, b) {
  return a + b
}

export function noCoverage() {
  return 'I do not have coverage'
}
