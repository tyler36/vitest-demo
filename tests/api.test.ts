import { test, expect } from 'vitest'
import { fetchTodoList } from '../src/api'

test('snapshot api', async () => {
  let data = await fetchTodoList()

  expect(data).toMatchSnapshot()
})

test('using dynamic data', () => {
  const user = {
    createdAt: new Date(),
    id: Math.floor(Math.random() * 20),
    name: 'LeBron James',
  }

  expect(user).toMatchSnapshot({
    createdAt: expect.any(Date),
    id: expect.any(Number),
  })
})
