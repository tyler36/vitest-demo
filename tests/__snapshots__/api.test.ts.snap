// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`snapshot api 1`] = `
{
  "completed": false,
  "id": 1,
  "title": "delectus aut autem",
  "userId": 1,
}
`;

exports[`using dynamic data 1`] = `
{
  "createdAt": Any<Date>,
  "id": Any<Number>,
  "name": "LeBron James",
}
`;
